core/Test/Tasty/Core.hs

```diff
@@ -243,7 +250,21 @@ data DependencyType
   | AllFinish
     -- ^ The current test tree will be executed after its dependencies finish,
     -- regardless of whether they succeed or not.
-  deriving (Eq, Show)
+  deriving (Eq, Show, Read)
+
+-- | Determines mode of execution of a 'TestGroupWithMode'
+data ExecutionMode
+  = Sequential DependencyType
+  -- ^ Execute tests one after another
+  | Parallel
+  -- ^ Execute tests in parallel
+  deriving (Show, Read)
+
+instance IsOption ExecutionMode where
+  defaultValue = Parallel
+  parseValue = readMaybe
+  optionName = Tagged "" -- internal option
+  optionHelp = Tagged "" -- internal option
```

Comment:
Did you check how `--help` looks now?

Reply:
Yeah, it doesn't print anything. But given that this is (now?) behavior we rely on I'll add a test.

Reply:
Ah wait, `IsOption` instances only get a corresponding command line flag if they implement `optionCLParser`. Otherwise they're internal anyway.

Reply:
I've added `mkInternalOptionCLParser` to make this much more explicit. Is this what you were after?

Reply:
`Test.Tasty.Options` is an exposed module, and I don't see much point in offering `mkInternalOptionCLParser` to users.

I meant literally `optionCLParser = mkOptionCLParser` [`internal`](https://hackage.haskell.org/package/optparse-applicative-0.17.0.0/docs/Options-Applicative.html#v:internal).

Reply:
Right, I wasn't aware of `internal`. Applied!